sklearn/tests/test_config.py

```diff
@@ -0,0 +1,26 @@
+from sklearn import get_config, set_config
+from sklearn.utils.testing import assert_equal
+
+
+def test_set_config():
+    assert_equal(get_config(), {'assume_finite': False})
+
+    # Not using as a context manager affects nothing
+    set_config(assume_finite=True)
```

Comment:
The benefit of ``set_config`` is that people will be looking for how to set the global state and having a public function is an easy to find (tab-complete) and document (references) way to do that.